domain/src/main/java/org/oppia/android/domain/translation/TranslationController.kt

```diff
@@ -285,14 +330,19 @@ class TranslationController @Inject constructor(
   }.build()
 
   private fun computeAppLanguage(
-    profileId: ProfileId,
-    systemLanguage: OppiaLanguage
+    systemLanguage: OppiaLanguage,
+    appLanguageSelection: AppLanguageSelection,
+    profileId: ProfileId
   ): OppiaLanguage {
-    val languageSelection = retrieveAppLanguageSelection(profileId)
-    return when (languageSelection.selectionTypeCase) {
-      AppLanguageSelection.SelectionTypeCase.SELECTED_LANGUAGE -> languageSelection.selectedLanguage
-      AppLanguageSelection.SelectionTypeCase.USE_SYSTEM_LANGUAGE_OR_APP_DEFAULT,
-      AppLanguageSelection.SelectionTypeCase.SELECTIONTYPE_NOT_SET, null -> systemLanguage
+    return if (profileId.internalId != USE_SYSTEM_LANGUAGE_INTERNAL_PROFILE_ID) {
```

Comment:
I'm not sure that I'm understanding why this special sentinel is needed. Could we just call getSystemLanguageLocale instead of getAppLanguage in the cases when we don't have a profile ID?

That being said I think we're actually supposed to be defaulting the app's language to the admin account's selection if it's available. I think that means you will need to create a new data provider returned by this controller that returns what's effectively the "default app language / locale" which first tries to default to the first admin account's selected app language or the system language if there is no admin profile or language selection. This would be the locale we fall back to if we don't have a profile ID or when we want to default to the system language.

Reply:
HI Ben this has been resolved with new method on TranslationController which now allows us to retrieve system languagelocale "getSystemLanguageLocale". Thanks.

Reply:
Had used this to previously get system language from "computeAppLanguage" to fall back to the same when user navigates to profile selection screen.